core/domain/exp_domain.py

```diff
@@ -3519,3 +3519,226 @@ def to_metadata_dict(self):
             'title': self.title,
             'objective': self.objective,
         }
+
+
+class StateIdMapping(object):
+    """Domain object for state id mapping model."""
+
+    def __init__(
+            self, exploration_id, exploration_version, state_name_to_ids,
+            latest_state_id_used):
+        """Initialize state id mapping model object.
+
+        Args:
+            exploration_id: str. The exploration id whose state names are
+            mapped.
+            exploration_version: int. The version of the exploration.
+            state_name_to_ids: dict. A dictionary mapping a state name to
+                unique ID.
+            latest_state_id_used: int. Last used state id.
```

Comment:
The largest integer so far that has been used as a state ID for this exploration.

(Want to distinguish between "largest" and "last".)

Reply:
Done